setup.sh

```diff
@@ -47,53 +47,51 @@ function check_dependencies()
   local pip_package_list=''
   local cmd=''
   local distro
-
+  declare -a debian=("dpkg-query -s " "apt install -y")
+  declare -a arch=("pacman -Ql " "pacman -S")
+  declare -a gentoo=("qlist -I " "emerge")
+  declare -a fedora=("rpm -q " "dnf install -y")
+  declare -A distro_commands=(['arch']=arch ['gentoo']=gentoo ['fedora']=fedora ['debian']=debian)
+  declare -n distro_cmd
   distro=$(detect_distro '/')
 
-  if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Ql "$package" &> /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
-  elif [[ "$distro" =~ 'debian' ]]; then
-    while IFS='' read -r package; do
-      installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/debian.dependencies"
-    cmd="apt install -y $package_list"
-  elif [[ "$distro" =~ 'fedora' ]]; then
+  for distro_cmd in ${distro_commands["$distro"]}; do
+    install_cmd="${distro_cmd[1]}"
+    list_pkgs="${distro_cmd[0]}"
```

Comment:
Maybe to keep this part of the code more readable, you can create two readonly global variables that replace the index with something more meaningful.